ui/views/mus/desktop_window_tree_host_mus.cc

```diff
@@ -842,8 +858,22 @@ void DesktopWindowTreeHostMus::OnActiveFocusClientChanged(
 void DesktopWindowTreeHostMus::OnWindowPropertyChanged(aura::Window* window,
                                                        const void* key,
                                                        intptr_t old) {
-  DCHECK_EQ(window, desktop_native_widget_aura_->content_window());
   DCHECK(!window->GetRootWindow() || this->window() == window->GetRootWindow());
+#if defined(USE_OZONE) && !(OS_CHROMEOS)
+  // In ozone, the widget's client and non-client views must be updated once the
+  // root window has its window state updated. Otherwise, once the host window
+  // manager updates the window state, the views are out-dated and do not
+  // correspond to actual state. So, update them here on every window state
+  // change happened with the root window. The behaviour of
+  // DesktopWindowTreeHostX11::OnWmStateChanged is actually copied here.
+  if (window == this->window()) {
+    if (key == aura::client::kShowStateKey)
+      Relayout();
+    return;
+  }
+#endif
+
+  DCHECK_EQ(window, desktop_native_widget_aura_->content_window());
```

Comment:
if DCHECK is out of the #if block. can it be upstreamed?

Reply:
this DCHECK has already been here. It was just moved a bit.